app/routes/planets_route.py

```diff
@@ -0,0 +1,96 @@
+from flask import Blueprint, abort, make_response, request, Response
+from ..db import db
+from app.models.planets import Planet
+
+planets_bp = Blueprint("planets_bp", __name__, url_prefix="/planets")
+
+@planets_bp.post("")
+def create_planet():
+    request_body = request.get_json()
+    name = request_body["name"]
+    description = request_body["description"]
+    size = request_body["size"]
+
+    new_planet = Planet(name=name, description=description, size=size)
+    db.session.add(new_planet)
+    db.session.commit()
+
+    response = new_planet.to_dict()
+    return response, 201
+
+@planets_bp.get("")
+def get_all_planets():
+    # query = db.select(Planet).order_by(Planet.id)
+    # planets = db.session.scalars(query)
+
+    # planets_response = [planet.to_dict() for planet in planets]
+    # return planets_response
+        query = db.select(Planet)
+        name_param = request.args.get("name")
+
+        if name_param:
+        # restrict to matching planet
+            query = query.where(Planet.name == name_param)
+    
+        description_param = request.args.get("description")
+        
+        if description_param:
+        # restrict to matching planet
+            query = query.where(Planet.description.ilike(f"%{description_param}%"))
+    
+        size_param = request.args.get("size")
+        if size_param:
+        # restrict to matching planet
+            query = query.where(Planet.ilike(f"%{size_param}%"))
+    
+        query = query.order_by(Planet.id)
```

Comment:
As you start to add more models, you may end up seeing code like this repeated across models (similar to a validate_id). I wonder if there is a way you could extract this into a helper function that could work across models?